### DISCUSSION TITLE
use just JS frameworks in Blazor Hybrid UI Design

### DISCUSSION THREAD

**Discussion by hamed65kz** [dotnet/aspnetcore#59667] [General]
I am currently on the verge of starting a project and need to choose the platform and framework that best align with the company’s human resources and technical requirements. One of the paths I am considering is Blazor Hybrid, and I wanted to consult with you to see if it fits my needs or not.

To begin, let me outline the project requirements:

This project is defined to refactor the architecture of an existing project. The current project is implemented with WPF, and the code is naturally in C#. Therefore, to utilize the existing code, the new project must also be able to use C# code.

This project will involve heavy processing and will be a resource-intensive application with a heavy UI that displays charts with a large number of points in real-time. So, performance is crucial.

I need to separate tasks and allow for parallel teamwork on UI and Bussiness Logic, which requires the use of the MVVM architecture or something similar.

The most important point: In our company, there is no expert in UI design with WPF, but we do have Web Front-end developers. Therefore, I was looking for a technology that could use C# on the Back-end while leveraging web technologies on the front-end, which is how I came across Blazor (or ASP.NET), a web development framework based on C#. Additionally, the Hybrid model will allow me to develop a Desktop App(which I am looking for).

I need the application to have Desktop and Tablet view mode.

Now, my question revolves mainly around point 4. Can I utilize our frontend developers, who work just with React or Vue.js, to develop the UI in Blazor Hybrid mode(without any razor syntax awareness)? Or do frontend developers need additional knowledge and must work with Razor syntax? If this approach is feasible, will it support the MVVM pattern and data binding?

**Comment by LucianoCBarilari** [comment]
Let me know if I understand you correctly. Do you want to use JavaScript interop in Blazor WASM, or are you planning to use React exclusively?

If you want to use React, you can create a project using the React and ASP.NET Core template. In this setup, the backend is handled by C# by default, and the frontend is React.

However, if you prefer using JavaScript interop, you can manage JavaScript in the Blazor frontend. The only drawback is that you'll need to implement an intermediate layer in C# to handle the communication between Blazor and JavaScript.

I hope this can help.

Docs for JS Interop
[JavaScript Interoperability in Blazor](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/?view=aspnetcore-8.0)
